rivertest/rivertest.go

```diff
@@ -29,8 +29,17 @@ type testingT interface {
 	Logf(format string, args ...any)
 }
 
-// Options for RequireInserted or RequireManyInserted including expectations for
-// various queuing properties that stem from InsertOpts.
+// Options for RequireInserted functions including expectations for various
+// queuing properties that stem from InsertOpts.
+//
+// When used with RequiredInserted or RequireInsertedMany, multiple properties
+// set on this struct increase the specifity on a job to match, acting like an
+// AND condition on each. So if multiple properties are set, a job must match
+// all of them to be considered a successful match.
+//
+// When used with RequireNotInserted, multiple properties act like an OR instead
+// of an AND. If an inserted job is found whose properties match any of the set
+// opts properties, a test failure is triggered.
```

Comment:
I don't this behavior is what I would expect here. My intuition (based on how I've used queue assertions for other libraries/ecosystems) is that I'm allowed to be as specific as I desire here, and that the assertion will ensure that no jobs have been inserted which match all of the conditions I specified. If I want, I can easily make sure that no jobs have been inserted on a certain queue, or if I want to be more specific, I can make sure that no jobs have been inserted on that queue and with that job type.

And in the affirmative case where I want to make sure a job _was_ inserted, it likewise matches _all_ of the criteria I specify, but ignores all those which I don't.